### PR TITLE
WASM support for Flutter web

### PR DESCRIPTION
**Pull request by amrgetment**
Adding WASM support to close
https://github.com/miguelpruivo/flutter_file_picker/issues/1480

Test by Flutter Beta channel or master channel and build by the following command
`flutter build web --wasm`

### CODE REVIEW

File: pubspec.yaml [hunk 1]
```diff
@@ -17,6 +17,8 @@ dependencies:
   path: ^1.8.2
   win32: ^5.1.1
   cross_file: ^0.3.3+7
+  js: ^0.7.1
```
Comment (IchordeDionysos) [inline pubspec.yaml:20]: Is that intended?
It's not used anymore and doesn't work with WASM anyways :)

**Review by IchordeDionysos — COMMENTED** [review]
Apart from the import it looks good!

Thanks for the PR 😍

File: pubspec.yaml (hunk 1 above)
Comment (amrgetment) [reply pubspec.yaml]: it is a mistake, I removed it I don’t use it anyway
thanks

**Review by IchordeDionysos — APPROVED** [review]

**Comment by amrgetment** [comment]
@IchordeDionysos You have to be a team member of this project to pass CI approval requirements 😁
Thanks for approval 😍🎯

**Comment by IchordeDionysos** [comment]
Yeah true :D But given I've looked over it, I figured it can't hurt :P

File: lib/_internal/file_picker_web.dart [hunk 2]
```diff
@@ -58,7 +60,9 @@ class FilePickerWeb extends FilePicker {
         Completer<List<PlatformFile>?>();
 
     String accept = _fileType(type, allowedExtensions);
-    InputElement uploadInput = FileUploadInputElement() as InputElement;
+    HTMLInputElement uploadInput =
```
Comment (navaronbracke) [inline lib/_internal/file_picker_web.dart:63]:
```suggestion
    HTMLInputElement uploadInput = HTMLInputElement();
```

The constructor calls document.createElement under the hood.

File: lib/_internal/file_picker_web.dart [hunk 3]
```diff
@@ -101,7 +105,12 @@ class FilePickerWeb extends FilePicker {
         }
       }
 
-      for (File file in files) {
+      for (var i = 0; i < files.length; i++) {
```
Comment (navaronbracke) [inline lib/_internal/file_picker_web.dart:108]:
```suggestion
      for (int i = 0; i < files.length; i++) {
```

File: lib/_internal/file_picker_web.dart [hunk 4]
```diff
@@ -144,14 +155,18 @@ class FilePickerWeb extends FilePicker {
...
+    uploadInput.addEventListener('change', changeEventListener.toJS);
+    uploadInput.addEventListener('cancel', cancelledEventListener.toJS);
 
     // Listen focus event for cancelled
-    window.addEventListener('focus', cancelledEventListener);
+    window.addEventListener('focus', cancelledEventListener.toJS);
 
     //Add input element to the page body
-    _target.children.clear();
+    var firstChild = _target.firstChild;
```
Comment (navaronbracke) [inline lib/_internal/file_picker_web.dart:165]: I'd prefer to use the real type instead of `var` here.

**Review by navaronbracke — CHANGES_REQUESTED** [review]
Some minor remarks

File: lib/_internal/file_picker_web.dart (hunk 2 above)
Comment (amrgetment) [reply lib/_internal/file_picker_web.dart]: refactored

File: lib/_internal/file_picker_web.dart (hunk 3 above)
Comment (amrgetment) [reply lib/_internal/file_picker_web.dart]: refactored

**Comment by amrgetment** [comment]
I added the change requests

**Review by navaronbracke — APPROVED** [review]

**Pull request merged**